File: 1erdepartamental/python/16.burbuja/main.py
"""
Programa: Burbuja
Demuestra algoritmo de ordenamiento burbuja con estadísticas
"""
#Clase Principal
from core.ordenador import OrdenadorBurbuja

#Funcion Principal
def main ():
    numeros = [64, 34, 25, 12, 22, 11, 90]

    print(f"Array original: {numeros}")
    ordenador = OrdenadorBurbuja()
    ordenador.ordenar(numeros)

    print(f"Array ordenado: {numeros}")
    print(f"Comparaciones: {ordenador.comparaciones}")
    print(f"Intercambios: {ordenador.intercambios}")

#Punto Pricipal
if __name__ == "__main__":
    main()

File: 1erdepartamental/c++/16.burbuja/main.cpp
#include <iostream>
#include "operations/OrdenadorBurbuja.h"

int main() {
    std::vector<int> numeros = {64, 34, 25, 12, 22, 11, 90};
    
    std::cout << "Array original: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    OrdenadorBurbuja<int> ordenador;
    ordenador.ordenar(numeros);
    
    std::cout << "Array ordenado: ";
    for (int n : numeros) std::cout << n << " ";
    std::cout << std::endl << std::endl;
    
    ordenador.mostrarEstadisticas();
    
    return 0;
}
